src/mongocxx/options/encrypt.cpp

```diff
@@ -27,14 +27,12 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace options {
 
 encrypt& encrypt::key_id(bsoncxx::types::b_binary key_id) {
-    _key_id = std::move(key_id);
+    // MNMLSTC on clang 3.6.2 reports an error when assigning to an optional of a struct.
+    _key_id = key_id;
+    _key_id_set = true;
```

Comment:
Fixes this warning:
```
[ 15%] Building CXX object src/mongocxx/CMakeFiles/mongocxx_shared.dir/options/encrypt.cpp.o
In file included from /home/travis/build/kevinAlbs/mongo-cxx-driver/src/mongocxx/options/encrypt.cpp:15:
In file included from /home/travis/build/kevinAlbs/mongo-cxx-driver/src/mongocxx/options/encrypt.hpp:19:
In file included from /home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/stdx/optional.hpp:22:
/home/travis/build/kevinAlbs/mongo-cxx-driver/build/src/bsoncxx/third_party/EP_mnmlstc_core-prefix/src/EP_mnmlstc_core/include/core/optional.hpp:313:7: error: 
      no viable conversion from 'bsoncxx::v_noabi::types::b_binary' to
      'bsoncxx::v_noabi::binary_sub_type'
      ::core::forward<Args>(args)...
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/travis/build/kevinAlbs/mongo-cxx-driver/build/src/bsoncxx/third_party/EP_mnmlstc_core-prefix/src/EP_mnmlstc_core/include/core/optional.hpp:252:36: note: 
      in instantiation of function template specialization
      'core::v1::optional<bsoncxx::v_noabi::types::b_binary>::emplace<bsoncxx::v_noabi::types::b_binary>'
      requested here
    if (not this->engaged) { this->emplace(::core::forward<T>(value)); }
                                   ^
/home/travis/build/kevinAlbs/mongo-cxx-driver/src/mongocxx/options/encrypt.cpp:30:13: note: 
      in instantiation of function template specialization
      'core::v1::optional<bsoncxx::v_noabi::types::b_binary>::operator=<bsoncxx::v_noabi::types::b_binary,
      void>' requested here
    _key_id = std::move(key_id);
            ^
In file included from /home/travis/build/kevinAlbs/mongo-cxx-driver/src/mongocxx/options/encrypt.cpp:15:
In file included from /home/travis/build/kevinAlbs/mongo-cxx-driver/src/mongocxx/options/encrypt.hpp:19:
In file included from /home/travis/build/kevinAlbs/mongo-cxx-driver/src/bsoncxx/stdx/optional.hpp:22:
/home/travis/build/kevinAlbs/mongo-cxx-driver/build/src/bsoncxx/third_party/EP_mnmlstc_core-prefix/src/EP_mnmlstc_core/include/core/optional.hpp:314:5: warning: 
      missing field 'size' initializer [-Wmissing-field-initializers]
    };
    ^
```

This appeared on clang 3.6.2 with MNMLSTC's optional (the bottom of the [logs in this Travis build](https://travis-ci.org/github/kevinAlbs/mongo-cxx-driver/jobs/699378458)). I did not dig too deeply into this, but I did observe that changing `b_binary`'s declaration from `struct` to `class` resolved this warning. So I suspect this is due to MNMLSTC's handling of optional with a `struct`.

I removed the getter, which seemed non-essential.

Reply:
Interesting.  I would like to keep the std::move, because if we are taking the key_id by l-value then it's only so that we can offer the caller the choice between a move and a copy.  Maybe we can work around whatever is happening in MNMLSTC by using optional's swap method:

`_key_id.swap(std::move(key_id))`

I think it's odd for us to not offer the getter.  Did the getter cause errors?  It is the driver's convention to offer getters like that for options classes.

I would really like to keep this as an optional instead of using a c-style value-bool pairing.  That is, after all, the whole point of optional!

Reply:
Ok, we can tackle the getter in that ticket later.